### DISCUSSION TITLE
Safari 16.3 autofill doesn't work with x-model

### DISCUSSION THREAD

**Discussion by dannyuk1982** [alpinejs/alpine#3482] [7. Bugs (archived)] [answered]
As per this CodePen, using Safari 16.3 autofill just sets values to null: https://codepen.io/dannyuk1982/pen/BaOvRmP

Under this code in x-model.js
```
if (event instanceof CustomEvent && event.detail !== undefined) {
  return typeof event.detail != 'undefined' ? event.detail : event.target.value
```

`event.detail` is `null`

Amending the code to this seems to fix it:
```
if (event instanceof CustomEvent && event.detail !== void 0) {
  return typeof event.detail != "undefined" && event.detail != null ? event.detail : event.target.value;
```

I did find several other bugs for Safari + Autofill, but all said it was fixed.

**Comment by ekwoka** [comment, marked as answer]
Probably just do

```js
if (event instanceof CustomEvent)
  return event.detail ?? event.target.value
```

`void 0` is just `undefined`, btw

edit: above is tested and works, so you can do that PR if you'd like, or I can

PR Exists: https://github.com/alpinejs/alpine/pull/3483

**Comment by dannyuk1982** [comment]
I'm not actually sure how the void got in there (multitasking this morning), only change should have been to add `&& event.detail != null` to the second line! Your code is more concise - please go ahead with the PR if you don't mind

**Comment by freinbichler** [comment]
This is a pretty big issue in the form I am currently building. It is basically unusable if the user uses Safari autocomplete to fill in fields.
@calebporzio it would be greatly appreciated if PR #3483 could be merged and released in a bugfix release ASAP.

**Comment by yabdab** [comment]
Safari has become the new IE

Hopefully @calebporzio  can address this soon.

**Comment by yabdab** [comment]
Until this gets properly fixed. which version should we roll-back to?

**Reply by ekwoka** [reply]
I believe the regression happened in 3.11, so 3.10.5 should be fine.

**Comment by dannyuk1982** [comment]
We already have a fix, I can't understand why it can't be rolled out? It's 2 lines!

**Comment by freinbichler** [comment]
I have since compiled Alpine from @ekwoka's PR #3483 and am using my custom build in production until the fix is released
